app/presenters/ht_registration_presenter.rb

```diff
@@ -52,13 +63,33 @@ def show_dsp_name
     action == :index ? link_to(dsp_name, ht_registration_path(id)) : dsp_name
   end
 
-  def show_env
-    return unless self[:env].present?
+  def show_detail_display_name
+    env["HTTP_X_SHIB_DISPLAYNAME"]
+  end
+
+  def show_detail_edu_person_principal_name
+    env["HTTP_X_SHIB_EDUPERSONPRINCIPALNAME"]
+  end
 
-    fields = JSON.parse self[:env]
-    return unless fields.any?
+  def show_detail_geoip
+    geoip = Services.geoip.city ip_address
+    [geoip.country.name, geoip.most_specific_subdivision.name, geoip.city.name].join(" — ")
+  rescue => _e
+    ""
+  end
+
+  def show_detail_organizations
```

Comment:
I don't think the whois parser is going to work for reverse lookups, sadly.

I think we do definitely want to be showing the value for HTTP_X_SHIB_IDENTITY_PROVIDER, mapped to an institution name through the institutions table.